src/flamenco/vm/fd_vm_interp_core.c

```diff
@@ -1131,18 +1131,18 @@
   if ( FD_UNLIKELY( ic_correction > cu ) ) err = FD_VM_ERR_SIGCOST; \
   cu -= fd_ulong_min( ic_correction, cu )
 
-sigtext:     err = FD_VM_ERR_SIGTEXT;  FD_VM_INTERP_FAULT;                     goto interp_halt;
-sigtextbr:   err = FD_VM_ERR_SIGTEXT;  /* ic current */      /* cu current */  goto interp_halt;
-sigsplit:    err = FD_VM_ERR_SIGSPLIT; FD_VM_INTERP_FAULT;                     goto interp_halt;
-sigcall:     err = FD_VM_ERR_SIGCALL;  /* ic current */      /* cu current */  goto interp_halt;
-sigstack:    err = FD_VM_ERR_SIGSTACK; /* ic current */      /* cu current */  goto interp_halt;
-sigill:      err = FD_VM_ERR_SIGILL;   FD_VM_INTERP_FAULT;                     goto interp_halt;
-sigsegv:     err = FD_VM_ERR_SIGSEGV;  FD_VM_INTERP_FAULT;                     goto interp_halt;
-sigcost:     err = FD_VM_ERR_SIGCOST;  /* ic current */      cu = 0UL;         goto interp_halt;
-sigsyscall:  /* err current */         /* ic current */      /* cu current */  goto interp_halt;
-sigfpe:      err = FD_VM_ERR_SIGFPE;   FD_VM_INTERP_FAULT;                     goto interp_halt;
-sigfpeof:    err = FD_VM_ERR_SIGFPE_OF;FD_VM_INTERP_FAULT;                     goto interp_halt;
-sigexit:     /* err current */         /* ic current */      /* cu current */  goto interp_halt;
+sigtext:     err = FD_VM_ERR_SIGTEXT;     FD_VM_INTERP_FAULT;                     goto interp_halt;
+sigtextbr:   err = FD_VM_ERR_SIGTEXT;     /* ic current */      /* cu current */  goto interp_halt;
+sigsplit:    err = FD_VM_ERR_SIGSPLIT;    FD_VM_INTERP_FAULT;                     goto interp_halt;
+sigcall:     err = FD_VM_ERR_SIGCALL;     /* ic current */      /* cu current */  goto interp_halt;
+sigstack:    err = FD_VM_ERR_SIGSTACK;    /* ic current */      /* cu current */  goto interp_halt;
+sigill:      err = FD_VM_ERR_SIGILL;      FD_VM_INTERP_FAULT;                     goto interp_halt;
+sigsegv:     err = FD_VM_ERR_SIGSEGV;     FD_VM_INTERP_FAULT;                     goto interp_halt;
+sigcost:     err = FD_VM_ERR_SIGCOST;     /* ic current */      cu = 0UL;         goto interp_halt;
+sigsyscall:  err = FD_VM_ERR_SIGSYSCALL;  /* ic current */      /* cu current */  goto interp_halt;
```

Comment:
I think that this changes the semantic? Syscalls return a typed error. If we change it to "syscall error", then we loose the type.

FWIW I think the SIMD is very unclear because they rely on SyscallError which is a rust thing, and it's also very unclear when it's thrown vs not. For example, memory violation inside a syscall... is that syscall error or ebpf!?

If we want to avoid syscalls (and I think we should), I think we need to store another var that says "error_because_of_syscall", not reuse the existing err.

Reply:
> If we want to avoid syscalls (and I think we should), I think we need to store another var that says "error_because_of_syscall", not reuse the existing err.

This `FD_VM_ERR_SIGSYSCALL` is a new VM error code specifically for the purpose you mentioned. Do we need a separate variable?

> FWIW I think the SIMD is very unclear because they rely on SyscallError which is a rust thing, and it's also very unclear when it's thrown vs not. For example, memory violation inside a syscall... is that syscall error or ebpf!?

Yeah it's a little confusing. I am taking it as "read the error code output of the VM execution and determine if it was caused by a syscall" situation

Reply:
Makes sense

Reply:
Yeah, it's about whether or not the error was thrown in the middle of a basic block or not. Syscall errors are thrown at boundaries of basic blocks, so we don't have to set consume all the requested in the failure cases.

> For example, memory violation inside a syscall... is that syscall error or ebpf!?

This is classified as a syscall error because of _where_ it was thrown. I do agree that the SIMD can be made clearer.